server/core/src/main/java/org/infinispan/server/core/backup/resources/CacheResource.java

```diff
@@ -122,56 +120,61 @@ public CompletionStage<Void> restore(ZipFile zip) {
       ConfigurationManager configurationManager = SecurityActions.getGlobalComponentRegistry(cm).getComponent(ConfigurationManager.class);
       Properties properties = new Properties();
       properties.put(CacheParser.IGNORE_DUPLICATES, true);
-      for (String cacheName : resources) {
-         CompletionStage<Void> cs = blockingManager
-               .supplyBlocking(() -> recoverCache(cacheName, properties, configurationManager, zip), "restore-cache-" + cacheName)
-               .thenCompose(Function.identity());
-         stages.dependsOn(cs);
-      }
-      return stages.freeze();
+      return CompletionStages.performConcurrently(resources, ProcessorInfo.availableProcessors(), blockingManager.asExecutor("cache-restore"), cacheName -> {
+         long start = System.currentTimeMillis();
```

Comment:
Thinking more should we do it per cache or for all of them together? And actually I am thinking we should probably just change these to be TRACE/DEBUG messages stating start and end of each cache instead maybe.

Reply:
A start/top message per cache seems good enough for debugging.